### PR TITLE
Exec on dest context with esdt transfer tests

### PR DESCRIPTION
**Pull request by sasurobert**
added a set of tests for interaction of NFTs and ESDTs with smart contract. Small fixes and refactors of code.

### CODE REVIEW

File: core/vmcommon/input.go [hunk 1]
```diff
@@ -18,6 +18,10 @@ const (
 	// AsynchronousCallBack means that an AsynchronousCall was performed
 	// previously, and now the control returns to the caller SmartContract's callBack method
 	AsynchronousCallBack
+
+	// ESDTTransferAndExecute means that there is a smart contract execution after the ESDT transfer
+	// this needed in order to skip the check whether a contract is payable or not
```
Comment (iulianpascalau) [inline core/vmcommon/input.go:23]: this is needed *

File: integrationTests/vm/esdt/esdtNFTSCs_test.go [hunk 2]
```diff
@@ -1 +1,331 @@
...
+		scAddress,
+		string(txData),
+		integrationTests.AdditionalGasLimit+core.MinMetaTxExtraGasCost,
+	)
+
+	time.Sleep(time.Second)
+	nonce, round = integrationTests.WaitOperationToBeDone(t, nodes, nrRoundsToPropagateMultiShard, nonce, round, idxProposers)
+	time.Sleep(time.Second)
+
+	txData = []byte("nftCreate" + "@" + hex.EncodeToString([]byte(tokenIdentifier)) +
```
Comment (iulianpascalau) [inline integrationTests/vm/esdt/esdtNFTSCs_test.go:70]: a job for tx data builder :D

File: core/vmcommon/input.go (hunk 1 above)
Comment (sasurobert) [reply core/vmcommon/input.go:23]: done

File: integrationTests/vm/esdt/esdtNFTSCs_test.go (hunk 2 above)
Comment (sasurobert) [reply integrationTests/vm/esdt/esdtNFTSCs_test.go]: I've got used to old style tx creation :dagger:

File: integrationTests/vm/esdt/esdtLocalFunsSC_test.go [hunk 3]
```diff
@@ -90,15 +90,123 @@ func TestESDTLocalMintAndBurnFromSC(t *testing.T) {
...
+		"@" + hex.EncodeToString(big.NewInt(100).Bytes()))
+	integrationTests.CreateAndSendTransaction(
+		nodes[0],
+		nodes,
+		big.NewInt(0),
+		scAddress,
+		string(txData),
+		integrationTests.AdditionalGasLimit,
+	)
+	integrationTests.CreateAndSendTransaction(
```
Comment (bogdan-rosianu) [inline integrationTests/vm/esdt/esdtLocalFunsSC_test.go:159]: why send the same tx twice?

File: integrationTests/vm/esdt/esdtLocalFunsSC_test.go [hunk 4]
```diff
@@ -90,15 +90,123 @@ func TestESDTLocalMintAndBurnFromSC(t *testing.T) {
...
+		"@" + hex.EncodeToString(big.NewInt(50).Bytes()))
+	integrationTests.CreateAndSendTransaction(
+		nodes[0],
+		nodes,
+		big.NewInt(0),
+		scAddress,
+		string(txData),
+		integrationTests.AdditionalGasLimit,
+	)
+	integrationTests.CreateAndSendTransaction(
```
Comment (bogdan-rosianu) [inline integrationTests/vm/esdt/esdtLocalFunsSC_test.go:184]: why send the same tx twice?

File: integrationTests/vm/esdt/esdtNFTSCs_test.go [hunk 5]
```diff
@@ -1 +1,429 @@
...
+	)
+	integrationTests.CreateAndSendTransaction(
+		nodes[0],
+		nodes,
+		big.NewInt(0),
+		scAddress,
+		string(txData),
+		integrationTests.AdditionalGasLimit,
+	)
+	integrationTests.CreateAndSendTransaction(
```
Comment (bogdan-rosianu) [inline integrationTests/vm/esdt/esdtNFTSCs_test.go:57]: why sending the same tx three times?

**Review by bogdan-rosianu — CHANGES_REQUESTED** [review]

File: integrationTests/vm/esdt/esdtLocalFunsSC_test.go (hunk 3 above)
Comment (sasurobert) [reply integrationTests/vm/esdt/esdtLocalFunsSC_test.go:159]: want to create more nft's - test that they are created with subsequent nonces, burn and transfer some of them

File: integrationTests/vm/esdt/esdtLocalFunsSC_test.go (hunk 4 above)
Comment (sasurobert) [reply integrationTests/vm/esdt/esdtLocalFunsSC_test.go:184]: want to create more nft's - test that they are created with subsequent nonces, burn and transfer some of them

File: integrationTests/vm/esdt/esdtNFTSCs_test.go (hunk 5 above)
Comment (sasurobert) [reply integrationTests/vm/esdt/esdtNFTSCs_test.go:57]: want to create more nft's - test that they are created with subsequent nonces, burn and transfer some of them

**Review by bogdan-rosianu — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**